feat(cli): amtinfo command add amt ipaddr and osipaddr

tongsean9807 opened the pull request:
Add os ipaddress to amtinfo command. 
amtinfo command now show amt ipaddrss and os ipaddress. 
os ipadress select by mapping to mac address from amt.

Resolves: #467

tongsean9807 commented:
Output:
![image](https://github.com/open-amt-cloud-toolkit/rpc-go/assets/105337968/aa99427b-d5d8-427b-afdd-d0c4c2087977)

rsdmike commented on pkg/pthi/commands.go:
```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
+		return 0, errors.New("fail")
+	}
+
+	zero_bytes := []byte{0, 0, 0, 0, 0, 0, 0}
+
+	if bytes.Equal([]byte(results), zero_bytes) {
+		return 0, nil
+	}
```
can be combines to:
``` suggestion
	// Check if the results array is all zeros
	if bytes.Equal(results, make([]byte, 7)) {
		return 0, nil
	}
```

rsdmike commented on pkg/pthi/commands.go:
```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
+	}
+
+	for _, iface := range interfaces {
+		if iface.Flags&net.FlagUp == 0 {
```
can be combined:
``` suggestion
	// Skip interfaces that are down or are loopback interfaces
		if iface.Flags&net.FlagUp == 0 || iface.Flags&net.FlagLoopback != 0 {
			continue
		}
```

rsdmike commented on pkg/pthi/commands.go:
```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
+		if bytes.Equal(hwaddr, []byte(results)) {
+			addrs, err := iface.Addrs()
+			if err != nil {
+				return 0, errors.New("fail")
```
ideally, a bit more descriptive error message would be good. For example "failed to get interface addresses"

rsdmike commented on pkg/pthi/commands.go:
```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
 	}
 }
 
+func GetOSIPAddress(results []uint8) (uint32, error) {
```
The PTHI package really is meant only for calls that interact with PTHI. Please move this to the internal folder. Along with that, we don't really need to convert the IP into a uint32. The end intention is display this on the command line so why bother converting it to uint32 and change the endian and all, just to convert it back? There no reason to emulate AMT in this regard since this data comes from the OS.

rsdmike reviewed (changes requested):
I like this being added. I left a few nit picks, but most importantly is to remove the uint32 conversion and move it out of the PTHI package and into the interal package, likely in the internal/local/info.go file.

rsdmike commented on pkg/pthi/commands.go:
```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
+
+		hwaddr := iface.HardwareAddr
+
+		if bytes.Equal(hwaddr, []byte(results)) {
```
[]byte(results) seems unnecessary as it is already passed in as a uint8 which is the same thing.

tongsean9807 commented:
Hi,

I have move the GetOSIPAddress() to internal/local/info.go
Let me know what other changes is needed.

I will squash them after fixing all the changes.

Thank you

rsdmike commented:
@tongsean9807 looks good to me. You think you can muster up a unit test or two?

rsdmike commented on internal/local/info.go:
```diff
@@ -158,28 +218,44 @@ func (service *ProvisioningService) DisplayAMTInfo() (err error) {
 			service.PrintOutput("DHCP Mode    		: " + wired.DHCPMode)
 			service.PrintOutput("Link Status  		: " + wired.LinkStatus)
-			service.PrintOutput("IP Address   		: " + wired.IPAddress)
+			service.PrintOutput("AMT IP Address   	: " + wired.IPAddress)
```
check spacing on the `:` to ensure its aligned

tongsean9807 replied:
change the space to tab

tongsean9807 commented:
Hi, 

I have changed the space to tab, which causes the misalignment when viewing in github.
I have added the unittest but I am unable to overwrite the net.interface. 
Let me know if there are better ways to improve the unittest.

Thank you

rsdmike commented:
Hi @tongsean9807 , Thank you for adding the unit test! I should have sent this to you, we have a `NetEnumerator` https://github.com/open-amt-cloud-toolkit/rpc-go/blob/bacb0ebc9fff12a92ccaeae5ddca8123f717fe2a/internal/flags/flags.go#L30-L31  that you can use to abstract the net.Interfaces calls. 

Can take a look at https://github.com/open-amt-cloud-toolkit/rpc-go/blob/bacb0ebc9fff12a92ccaeae5ddca8123f717fe2a/internal/flags/flags_test.go#L115 for an example of how it is used in tests.

For the code that uses the `NetEnumerator` check out: https://github.com/open-amt-cloud-toolkit/rpc-go/blob/bacb0ebc9fff12a92ccaeae5ddca8123f717fe2a/internal/flags/maintenance.go#L196.

tongsean9807 commented:
Hi,

I have add positive and negative unit test. 
And use the net enumerator from flags package.

Thank you for the pointing out the net enumerator.

Regards,
Sean

tongsean9807 commented:
Hi @rsdmike,

I hope you're doing well. I want to follow up on this PR.
I have made the changes.
Have you had a chance to review it?

Thank you

tongsean9807 reviewed (commented):
I have move getosipaddr func to info.go

rsdmike commented:
> Hi @rsdmike,
> 
> I hope you're doing well. I want to follow up on this PR. I have made the changes. Have you had a chance to review it?
> 
> Thank you

Thanks for following up, looks good to me! We'll get it merged.
